[CIVIC-807] Updated Radio component in Storybook.

govindmaloo opened the pull request:
https://salsadigital.atlassian.net/browse/CIVIC-807

## Checklist before requesting a review

- [x] I have formatted the subject to include ticket number as `[CIVIC-123] Verb in past tense with dot at the end.`
- [x] I have added a link to the JIRA ticket
- [x] I have provided information in `Changed` section about WHY something was done if this was not a normal implementation
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have run new and existing relevant tests locally with my changes, and they passed
- [x] I have provided screenshots, where applicable

## Changed
1. Added Radio and Checkbox story
2. Updated color as per figma.

## Screenshots

<img width="1070" alt="Screenshot 2022-08-28 at 10 19 39 PM" src="https://user-images.githubusercontent.com/4437018/187085418-51217522-1246-4a33-8094-d60b56c38488.png">

<img width="1079" alt="Screenshot 2022-08-28 at 10 20 53 PM" src="https://user-images.githubusercontent.com/4437018/187085436-39c9a884-eadc-425e-a142-b224210fd1de.png">

AlexSkrypnyk commented on docroot/themes/contrib/civictheme/civictheme_library/components/00-base/_variables.components.scss:
```diff
@@ -430,12 +430,14 @@ $civictheme-input-dark-disabled-background-color: civictheme-color('neutral', 6)
-$civictheme-radio-light-border-color: civictheme-color('neutral', 6) !default;
+$civictheme-radio-light-border-color: civictheme-color('primary', 5) !default;
 $civictheme-radio-light-background-color: civictheme-color('neutral', 3) !default;
+$civictheme-radio-light-shadow-color: civictheme-color('neutral', 1) !default;
```
these variable names are named after the purpose, not the CSS property. In this case - it is a background behind the radio circle, so the variable would be `$civictheme-radio-light-background-color`.

AlexSkrypnyk commented on docroot/themes/contrib/civictheme/civictheme_library/components/01-atoms/checkbox/checkbox.stories.js:
```diff
@@ -0,0 +1,32 @@
+  const generalKnobTab = typeof knobTab === 'string' ? knobTab : 'General';
+
+  const generalKnobs = {
+    theme: radios(
```
not all of the props were mapped to the props in twig file

and some of the props were not removed from the twig file

AlexSkrypnyk reviewed (changes requested):
@govindmaloo 
please see inline comments.

I will update the ticket ACs to capture things better

AlexSkrypnyk commented:
@govindmaloo 
Please separate changes for each ticket in a separate PR going forward so we could merge independently. Thank you

AlexSkrypnyk commented on docroot/themes/contrib/civictheme/civictheme_library/components/01-atoms/checkbox/checkbox.twig:
```diff
@@ -18,9 +20,13 @@
 {% set state_class = 'civictheme-input--%s'|format(state) %}
```
`state` is no longer used - please remove

AlexSkrypnyk commented on docroot/themes/contrib/civictheme/civictheme_library/components/01-atoms/radio/radio.twig:
```diff
@@ -5,10 +5,12 @@
  *
  * Variables:
  * - theme: [string] Theme: light, dark.
+ * - text: [string] Input field label.
```
could you please change `text` to `content` - sorry, I wrote incorrectly in the ticket - updated now

AlexSkrypnyk commented on docroot/themes/contrib/civictheme/civictheme_library/components/01-atoms/radio/radio.twig:
```diff
@@ -19,9 +21,13 @@
-<div class="civictheme-radio">
+<div class="civictheme-radio {{ theme_class }}">
   <input class="civictheme-radio__element civictheme-input__element {{ modifier_class }}" type="radio" value="{{ value }}" {% if attributes is not empty %}{{ attributes|raw }}{% endif %} {{ disabled ? 'disabled' }} {{ required ? 'required' }}/>
   <span class="civictheme-radio__state"></span>
```
is this still used?

AlexSkrypnyk commented on docroot/themes/contrib/civictheme/civictheme_library/components/01-atoms/checkbox/checkbox.twig:
```diff
@@ -18,9 +20,13 @@
-<div class="civictheme-checkbox">
+<div class="civictheme-checkbox {{ theme_class }}">
   <input class="civictheme-checkbox__element civictheme-input__element {{ modifier_class }}" type="checkbox" value="{{ value }}" {% if attributes %}{{ attributes|raw }}{% endif %} {{ disabled ? 'disabled' }} {{ required ? 'required' }}/>
   <span class="civictheme-checkbox__state"></span>
```
is this still used?

AlexSkrypnyk commented on docroot/themes/contrib/civictheme/civictheme_library/components/01-atoms/checkbox/checkbox.twig:
```diff
@@ -5,9 +5,11 @@
  *
  * Variables:
  * - theme: [string] Theme: light, dark.
+ * - text: [string] Input field label.
```
could you please update `text` to `content` - sorry, I wrote incorrectly in the ticket - fixed now

AlexSkrypnyk commented on docroot/themes/contrib/civictheme/civictheme_library/components/01-atoms/checkbox/checkbox.scss:
```diff
@@ -5,7 +5,11 @@
 @import '../../00-base/mixins/form';
```
do we need this mixin?

govindmaloo replied:
Yes we are using the same mixin for the element CSS.

AlexSkrypnyk commented on docroot/themes/contrib/civictheme/civictheme_library/components/01-atoms/checkbox/checkbox.scss:
```diff
@@ -5,7 +5,11 @@
+  display: flex;
+  align-items: center;
 
   &__element {
```
do we need this element?

govindmaloo replied:
Yes, We are applying CSS classes to input element same as before.

AlexSkrypnyk commented on docroot/themes/contrib/civictheme/civictheme_library/components/01-atoms/radio/radio.scss:
```diff
@@ -5,11 +5,39 @@
 @import '../../00-base/mixins/form';
```
do we need this mixin?

AlexSkrypnyk commented on docroot/themes/contrib/civictheme/civictheme_library/components/01-atoms/radio/radio.scss:
```diff
@@ -5,11 +5,39 @@
+  align-items: center;
 
   .civictheme-radio__element {
     @include civictheme-checkbox-radio('radio');
```
why do we need this mixin instead of just having rules here?

govindmaloo replied:
Do we need to separate both component CSS without using mixins?

AlexSkrypnyk commented on docroot/themes/contrib/civictheme/civictheme_library/components/01-atoms/radio/radio.scss:
```diff
@@ -5,11 +5,39 @@
+    #{$root}__label {
+      color: $civictheme-radio-label-light-color;
+
+      @include civictheme-typography('text-regular');
```
please do not use non-colour mappings in the colo-related specific rules.  Please move this to the outside of `.civictheme-theme-*` rule

AlexSkrypnyk reviewed (changes requested):
please see inline comments

AlexSkrypnyk commented on docroot/themes/contrib/civictheme/civictheme_library/components/01-atoms/checkbox/checkbox.twig:
```diff
@@ -5,22 +5,26 @@
+{% set modifier_class = '%s %s %s %s %s'|format(theme_class, disabled_class, required_class, error_class, modifier_class|default('')) %}
 
-<div class="civictheme-checkbox">
+<div class="civictheme-checkbox {{ theme_class }}">
```
please set `modifier_class` on the outer `div` and re-style accordingly

AlexSkrypnyk commented on docroot/themes/contrib/civictheme/civictheme_library/components/01-atoms/radio/radio.twig:
```diff
@@ -5,23 +5,27 @@
+{% set modifier_class = '%s %s %s %s %s'|format(theme_class, disabled_class, required_class, error_class, modifier_class|default('')) %}
 
-<div class="civictheme-radio">
+<div class="civictheme-radio {{ theme_class }}">
```
please set `modifier_class` on the outer `div` and re-style accordingly

govindmaloo commented:
@AlexSkrypnyk I have updated both components individually so no form mixin is required. Actioned other items and added separate required variables for both component. Now, these 2 component is separated and the user can directly use them as individual component.

govindmaloo commented:
@AlexSkrypnyk Please confirm if I need to remove form.scss completely.

AlexSkrypnyk reviewed (changes requested):
@govindmaloo 
Please test all combinations of props.

The Check is removed from the Checkbox when it is disabled. The check still should be present for checked disabled checkbox.

AlexSkrypnyk commented:
Also, please that the same is fixed for a radio

AlexSkrypnyk commented:
Could you please also update `Modifier class` label to `Additional classes`

govindmaloo commented:
> @govindmaloo Please test all combinations of props.
> 
> The Check is removed from the Checkbox when it is disabled. The check still should be present for checked disabled checkbox.

I have checked again and the style is working fine. If disabled user can't really click on it so need to use additional attributes checked along with disabled.

<img width="1112" alt="Screenshot 2022-08-31 at 7 03 31 AM" src="https://user-images.githubusercontent.com/4437018/187573377-e3a251a3-b293-49d0-b078-8e18a3d6eb44.png">
<img width="1120" alt="Screenshot 2022-08-31 at 7 05 04 AM" src="https://user-images.githubusercontent.com/4437018/187573450-2d2904b4-9e17-4dc9-a51b-9e4772a5f0a2.png">

govindmaloo commented:
Added new PR https://github.com/salsadigitalauorg/civictheme_source/pull/603 for checkbox component and removed from this one
